<a href="https://colab.research.google.com/github/erick-huang15/Python_Exercises/blob/main/Find_average_height_per_individual_olympian_per_year_exercise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

data = '/content/drive/MyDrive/CD: Week 3: Exploratory Viz/athleteEventsNoPersonal.csv'
olympian_data = pd.read_csv(data)
olympian_data

ID   Age  ...                                     Event Medal
0           5  21.0  ...          Speed Skating Women's 500 metres   NaN
1           5  21.0  ...        Speed Skating Women's 1,000 metres   NaN
2           5  25.0  ...          Speed Skating Women's 500 metres   NaN
3           5  25.0  ...        Speed Skating Women's 1,000 metres   NaN
4           5  27.0  ...          Speed Skating Women's 500 metres   NaN
...       ...   ...  ...                                       ...   ...
40611  135567  28.0  ...    Rowing Men's Lightweight Coxless Fours   NaN
40612  135570  27.0  ...  Ski Jumping Men's Large Hill, Individual   NaN
40613  135570  27.0  ...        Ski Jumping Men's Large Hill, Team   NaN
40614  135571  30.0  ...                      Bobsleigh Men's Four   NaN
40615  135571  34.0  ...                      Bobsleigh Men's Four   NaN

[40616 rows x 12 columns]

In [3]:
num_row = len(olympian_data) #this is to get the number of rows
print(f'There are a total of',num_row,'rows')

There are a total of 40616 rows


In [4]:
num_col = len(olympian_data.columns) #this is to get the number of columns
num_col
print(f'Theree are a total of',num_col,'columns')

Theree are a total of 12 columns


### a. How many rows? 
#### There are 40616 rows

### b. How many columns? 
#### There are 12 columns

### c. Which columns have missing values? (Why do think there are there so many null values in the Medals column?) 
#### Age, Height, Medal all have missing values. There are so many missing values in medals column because those were all participants who did not win any metals.

In [ ]:
olympian_data.isna().sum()

ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64


### d. How many entries correspond to the city of London?
#### There are a total of 3370 entires correspondence to the city of London

In [ ]:
london_filter = olympian_data['City']=='London' # filter london
london_data = olympian_data[london_filter] # store the filter in a dataframe
london_data['ID'].count() # count number of ID to get the number of entries 

3370

### e. What age is the youngest athlete in our sample data?  Hint: use min()
### The youngest olympian is 11 years old

In [ ]:
olympian_data['Age'].min() 

11.0

### Create filters and find out
### a. How many athletes who participated in the Sport "Tug-of-War" were from Team "Sweden" in our sample dataset
#### There are a total of 6 Sweden athletes that participated in the Tug-of-War sports

In [17]:
tow_filter = olympian_data['Sport']== 'Tug-Of-War' # this is to filter the sport Tug-Of-War 
tow_data = olympian_data[tow_filter] # this is to store the filter in a dataframe
sweden_filter = tow_data['Team']=='Sweden' # this is to filter team Sweden from the Tug-Of-War dataframe
sweden_tow_data = tow_data[sweden_filter] # this is to store the team Sweden filter in a dataframe
sweden_tow_data['ID'].value_counts() # this is to count the number of unique ID that participated the tug-of-war sport on the Sweden team

34205    1
43105    1
44087    1
55169    1
64266    1
68647    1
Name: ID, dtype: int64

In [16]:
sweden_tow_data # this is to print the sweden team members that participated in tug-of-war

ID   Age  Height  ...       Sport                        Event   Medal
10023  34205  22.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
12681  43105  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
13015  44087  29.0   182.0  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze
16250  55169  23.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
19048  64266  21.0     NaN  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN
20550  68647  26.0   190.0  ...  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze

[6 rows x 12 columns]

In [18]:
olympian_data.head()

ID   Age  Height  ...          Sport                               Event Medal
0   5  21.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
1   5  21.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
2   5  25.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
3   5  25.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
4   5  27.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN

[5 rows x 12 columns]

### b. How many Gold medals were awarded to athletes over 40 years old? 
#### A total of 66 gold medals were awarded to athletes over 40 years old

In [69]:
age_filter  = olympian_data['Age']>40 # this is to create a filter for athletes greater than 40 years old
age_data = olympian_data[age_filter]
age_data.head()

ID   Age  ...                                              Event Medal
122  389  45.0  ...  Art Competitions Mixed Architecture, Designs F...   NaN
123  389  45.0  ...  Art Competitions Mixed Architecture, Architect...   NaN
131  432  42.0  ...           Canoeing Men's Kayak Singles, 500 metres   NaN
132  432  42.0  ...         Canoeing Men's Kayak Singles, 1,000 metres   NaN
186  677  55.0  ...                    Art Competitions Mixed Painting   NaN

[5 rows x 12 columns]

In [70]:
gold_medal_filter = age_data['Medal']=='Gold'
gold_medal_data = age_data[gold_medal_filter]
gold_medal_data.head()

ID   Age  ...                                        Event Medal
740    2735  41.0  ...  Shooting Men's Rapid-Fire Pistol, 25 metres  Gold
1103   4029  46.0  ...                              Polo Men's Polo  Gold
1421   4993  41.0  ...                      Sailing Mixed 10 metres  Gold
3040  10769  49.0  ...          Sailing Mixed Three Person Keelboat  Gold
3270  11617  44.0  ...                       Sailing Mixed 7 metres  Gold

[5 rows x 12 columns]

In [71]:
gold_medal_data['Medal'].value_counts() # this is to count the number of gold medals in the dataset

Gold    66
Name: Medal, dtype: int64

### Determine the average height of: 
### a. All rows included: 175.38
### b. All rows included, but grouped by Event
### c. Bonus: This dataset contains many athletes more than once (if they competed in multiple events and/or multiple years). You can confirm this by finding  .value_counts on the ID column!  For the bonus task we want to get the average height of all  the athletes, but don't want to include any athlete more than once.  Hint: We learned how to drop duplicates if EVERY column was identical, but now we want to drop a row any time just the ID is repeated.  To do this,  you can use a subset of your data as an argument in the drop_duplicates function.  You can also tell Python which of the duplicates to keep by using the keep argument.

In [ ]:
#olympian_data.head()
olympian_data['Height'].mean() # this is to get the average height

175.3811165585956

In [ ]:
olympian_data.groupby('Event')['Height'].mean() # this is to get the average olypian height of every events

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64